### DISCUSSION TITLE
Method-Style Access Getter

### DISCUSSION THREAD

**Discussion by max5432112345** [vuejs/pinia#1569] [Help and Questions] [answered]
### Reproduction

-

### Steps to reproduce the bug

In Vuex

Method-Style Access [#](https://vuex.vuejs.org/guide/getters.html#method-style-access)
You can also pass arguments to getters by returning a function. This is particularly useful when you want to query an array in the store:
```
getters: {
  // ...
  getTodoById: (state) => (id) => {
    return state.todos.find(todo => todo.id === id)
  }
}


store.getters.getTodoById(2) // -> { id: 2, text: '...', done: false }
```

How to do that with Pinia

```
const getItemsByType = computed((element: string) => {
  return data.value[element] ? data.value[element] : []
})
```

That not work well 
> Invalid prop: type check failed for prop "product". Expected Object, got Function 
```
  const getItemsByType = (element: string) => {
    return computed((element: string) => {
      return data.value[element] ? data.value[element] : []
    })
  }
```

And in that case is like action and not getter (computed)?

### Expected behavior

-

### Actual behavior

-

**Comment by max5432112345** [comment]
And that hot help

```
const fullName = computed({
  // getter
  get() {
    return firstName.value + ' ' + lastName.value
  },
  // setter
  set(newValue) {
    // Note: we are using destructuring assignment syntax here.
    [firstName.value, lastName.value] = newValue.split(' ')
  }
})

```

**Comment by max5432112345** [comment]
That work, but it is action and not getter, and actions shouldn't return something.
``
const getItemsByType = (element: string) => {
  return data.value[element] ? data.value[element] : []
}
``

**Reply by posva** [reply]
actions can return anything they want 🙂

**Comment by posva** [comment, marked as answer]
It's actually the same as Vuex:

Options stores:

```ts
defineStore('store', {
  getters: {
    foo: state => (param) => state.bar + param
  }
})
```

Setup Stores:

```ts
const getItemsByType = computed(() => 
  (element: string) => data.value[element] ? data.value[element] : []
)
```

**Reply by Liwoj** [reply]
What is the point of creating `computed` returning function? I mean, is there any advantage compared to using simple function ?

**Reply by posva** [reply]
In this example no. You can cache values outside of the returned function, that’s about it

**Reply by NukeTheMoon** [reply]
when you write a pinia store with a computed that returns a function, @nuxt/devalue prints warnings to console:

`WARN Cannot stringify a function`
